### PR TITLE
Helm chart

### PR DESCRIPTION
**Pull request by heywoodlh**
## Proposed changes

This adds a Helm chart so that Nuclei can be deployed easily in a Kubernetes cluster.

## Checklist

I don't think that testing is super essential with these changes, but I suppose it could be implemented with `helm template .`. If there is interest to implement this as a test I can add it.

- [x] Pull request is created against the [dev](https://github.com/projectdiscovery/nuclei/tree/dev) branch
- [ ] All checks passed (lint, unit/integration/regression tests etc.) with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by heywoodlh** [comment]
Documentation PR is here: https://github.com/projectdiscovery/nuclei-docs/pull/67

**Comment by heywoodlh** [comment]
Some things I'd like to note:

 - Using the Helm chart as its base, I was able to write a Helm chart at my place of employment that automatically scans all our web applications in our cluster. I think Nuclei's flexibility puts it in a unique position to fill the need of network-level vulnerability scanning and fit very nicely into Kubernetes. I have found that most Kubernetes-focused vulnerability scanners only focus on the OS and package level within containers. Nuclei could very much fill that gap in a lot of Kubernetes deployments, imo.
 - The Helm chart at the moment has no method of persistence for Nuclei templates. So the Helm chart is wholly dependent upon the community templates. I'll probably update the chart with this functionality sometime this week.
 - Something like [Mend Renovate](https://www.mend.io/free-developer-tools/renovate/) should be implemented to automatically keep the container image tags up to date in the values.yaml file

File: helm/templates/hpa.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{ include "nuclei.fullname" . }}
+  labels:
+    {{- include "nuclei.labels" . | nindent 4 }}
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: {{ include "nuclei.fullname" . }}
```
Comment (ShubhamRasal) [inline helm/templates/hpa.yaml:12]: can you please updae the name from `{{ include "nuclei.fullname" . }}` to `{{ include "nuclei.fullname" . }}-interactsh` ?
interactsh deployment name is wrong here

**Review by ShubhamRasal — CHANGES_REQUESTED** [review]

**Review by ShubhamRasal — APPROVED** [review]

**Comment by ehsandeep** [comment]
@heywoodlh thank you for sharing this.

**Pull request merged**